LaTeX/TeX.sublime-syntax

```diff
@@ -111,6 +118,54 @@ contexts:
       captures:
         1: punctuation.definition.backslash.tex
 
+  registers:
+    - match: (\\)({{registers}}){{endcs}}
+      captures:
+        1: punctuation.definition.backslash.tex
+        2: storage.type.tex
+      push: register-id
+
+  register-id:
+    - meta_scope: meta.register.tex
+    - match: \d+
+      scope: meta.number.integer.decimal constant.numeric.value.tex
+      pop: 1
+    - match: (\\){{cmdname}}
+      captures:
```

Comment:
Is this the unscoped `\macro` you are concerned about?

We could add a proper `scope` here, right?

Reply:
yes, I'm just not sure which scope. Maybe `variable.parameter.tex`?

Reply:
`variable.parameter` is normally used to scope argument names in function definitions.

I'd try to find a scope name which fits to `\def\five{0}`, so rules for goto definition vs. goto reference can be declared.

`\five` being scoped `meta.function entity.name.definition` seems a bit weird to me.

Maybe we should decide whether we want to scope macros `meta.function entity.name.function` or just `meta.macro entity.name.macro`

The latter one would probably be more generic in case macros are used for constants as well, which can't be distinguished by static syntax rules.

The proper counter part for 
- `entity.name.function` is `variable.function`. 
- `entity.name.macro` may be `constant.other.macro` (used in Erlang)

Reply:
we could also, at least for now, decide to not scope this directly, but go back to the main scope and let `general-commands` handle this. Given that I think such constructs will be extremely unlikely in actual user code, that would probably be good enough. Still, there remains the question of how to scope macros in general. At least for the language-provided ones, I think it would be nice to have separate scopes depending on whether they do take an argument or not, but you are right that we in general cannot determine this from the syntax alone.

Reply:
Agree, would probably enough. If we would like to maintain `meta.register` `general-commands` could be included in `register-id` replacing the `- match: (\\){{cmdname}}` pattern.

I'd be fine with both.

Reply:
yes, let's do that for now.